modules/configuring-cert-based-auth-quay-cloudsql.adoc

```diff
@@ -0,0 +1,195 @@
+$ oc create secret generic postgresql-client-certs \
+--from-file config.yaml=<path/to/config.yaml> <1>
+--from-file=tls.crt=<path/to/ssl_client_certificate.pem> <2>
+--from-file=tls.key=<path/to/ssl_client_key.pem> <3>
+--from-file=ca.crt=<path/to/ssl_server_certificate.pem> <4>
+----
+<1> Where` <config.yaml>` is your `base64 decoded` `config.yaml` file.
+<2> Where `ssl_client_certificate.pem` is your SSL certificate in `.pem` format.
+<3> Where `ssl_client_key.pem` is your SSL key in `.pem` format.
+<4> Where `ssl_server_certificate.pem` is your SSL root CA in `.pem` format.
+
+. Depending on your security requirements and the type of certificates that you are using, you can configure {productname} to verify the certificate against a trusted CA using the `verify-ca` SSL mode. Alternatively, you can configure {productname} to verify the certificate against a trusted CA *and* that the hostname matches a SAN entry in your certificates. Use one of the following verification methods.
+
+.. To verify the certificate against a trusted CA only, use the `verify-ca` SSL mode. Edit your `config.yaml` file to include the following information:
++
+[IMPORTANT]
+====
+The information included in the `DB_CONNECTION_ARGS` variable, for example, `sslmode`, `sslrootcert`, `sslcert`, and `sslkey` *must* match the information appended to the `DB_URI` variable. Failure to match might result in a failed connection.
+====
++
+[source,yaml]
+----
+DB_CONNECTION_ARGS:
+    autorollback: true
+    sslmode: verify-ca <1>
+    sslrootcert: /.postgresql/<root.crt> <2>
+    sslcert: /.postgresql/<cloudsql.crt> <3>
+    sslkey: /.postgresql/<cloudsql.key> <4>
+    threadlocals: true <5>
+DB_URI: postgresql://<dbusername>:<dbpassword>@34.29.130.58:5432/quay14?sslmode=verify-ca&sslcert=/.postgresql/<cloudsql.crt>&sslkey=/.postgresql/<cloudsql.key>&sslrootcert=/.postgresql/<root.crt> <6>
+----
+<1> Ensures that the database only tries an SSL/TLS connection and verifies that the server certificate is issued by a trusted CA.
+<2> The path of the `root.crt` file, which contains the root certificate used to verify the SSL/TLS connection with your CloudSQL database. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<3> The path of the `cloudsql.crt` file, which contains the client certificate used to authenticate the connection to your CloudSQL database. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<4> The path of the `cloudsql.key` file, which contains the private key associated with the client certificate. This file is mounted in the `Quay` pod from the Kubernetes secret.
+<5> Enables auto-rollback for connections.
+<6> The URI that accesses your CloudSQL database appended with your `root.crt`, `cloudsql.crt`, and `cloudsql.key` files. If using `verify-ca` as your SSL mode, you must also include the database username and password. The SSL/TLS information included in `DB_URI` must match the information provided in `DB_CONNECTION_ARGS`.
```

Comment:
"If using `verify-ca` as your SSL mode, you must also include the database username and password" -> no, this is not related with `verify-ca`, this is GCP CloudSQL specific: it requires database username and password in `DB_URI`

Reply:
Fixed to 

<6> The URI that accesses your CloudSQL database appended with your `root.crt`, `cloudsql.crt`, and `cloudsql.key` files. The SSL/TLS information included in `DB_URI` must match the information provided in `DB_CONNECTION_ARGS`. If you are using CloudSQL, you must include your database username and password in this variable.